src/platform/unix.rs

```diff
@@ -221,73 +257,91 @@ pub trait WindowExtUnix {
 
 impl WindowExtUnix for Window {
     #[inline]
+    #[cfg(feature = "x11")]
     fn xlib_window(&self) -> Option<raw::c_ulong> {
         match self.window {
             LinuxWindow::X(ref w) => Some(w.xlib_window()),
+            #[cfg(feature = "wayland")]
             _ => None,
         }
     }
 
     #[inline]
+    #[cfg(feature = "x11")]
     fn xlib_display(&self) -> Option<*mut raw::c_void> {
         match self.window {
             LinuxWindow::X(ref w) => Some(w.xlib_display()),
+            #[cfg(feature = "wayland")]
             _ => None,
         }
     }
 
     #[inline]
+    #[cfg(feature = "x11")]
     fn xlib_screen_id(&self) -> Option<raw::c_int> {
         match self.window {
             LinuxWindow::X(ref w) => Some(w.xlib_screen_id()),
+            #[cfg(feature = "wayland")]
             _ => None,
         }
     }
 
     #[inline]
     #[doc(hidden)]
+    #[cfg(feature = "x11")]
     fn xlib_xconnection(&self) -> Option<Arc<XConnection>> {
         match self.window {
             LinuxWindow::X(ref w) => Some(w.xlib_xconnection()),
+            #[cfg(feature = "wayland")]
             _ => None,
         }
     }
 
     #[inline]
+    #[cfg(feature = "x11")]
     fn set_urgent(&self, is_urgent: bool) {
+        #[allow(irrefutable_let_patterns)]
```

Comment:
This bumps minimum rust version to `1.43.0`, please, remove it for now, otherwise it's nearly impossible to ship winit apps, like alacritty. You are free to add a comment that it should be added in a future back.

Reply:
I assume you meant 1.33. But i'll change it.

Reply:
No, I mean exactly 1.43, since complication is failing on 1.43.